### PR TITLE
Chris901003 proposal for PoissonNet

### PR DESCRIPTION
**Pull request by chris901003**
---
A good PR should include the following parts:

- [x] Provide a clear title
- [x] Provide meaningful descriptions in the PR and the commit logs
- [x] Name your branch like ``<github-user-name>-hw#``, e.g., ``myname-hw2-attempt1``
- [x] Merge from your specific branch (***not master***) to the `hw*` branch (***not master***, again)
- [x] Exclude irrelevant files, such as binaries, text files, or dot files
- [x] Make sure tests/CI pass

### CODE REVIEW

File: project/chris901003/README.rst [hunk 1]
```diff
@@ -0,0 +1,103 @@
+Basic Log Analysis
+****************************
+
+This project is to analyze log files.
+
+****************************
+Basic Information
+****************************
+
+**Project**: BasicLogAnalysis
```
Comment (yungyuc) [inline project/chris901003/README.rst]: The name is simple and clear.  I like it.

But the topic may not give you a long mileage in the final presentation.  Log processing does not usually use numerical calculation.

File: project/chris901003/README.rst [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+
+**Project**: BasicLogAnalysis
+
+**GitHub**: https://github.com/chris901003/BasicLogAnalysis
+
+****************************
+Problem to Solve
+****************************
+
+The main objective of this project is to address the challenge of quickly categorizing large volumes of log information,
```
Comment (yungyuc) [inline project/chris901003/README.rst]: How large are the data you plan to process?  If we are talking about tera-byte-scale data, you will need a sophisticated I/O implementation.  The computation does not involve a lot of numerical calculation for your presentation.

File: project/chris901003/README.rst [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+* Week 3 (09/16): Proposal Submission
+* Week 4 (09/23): Setup Development Environment
+* Week 5 (09/30): Implement FastText Sentence Vectorization
+* Week 6 (10/07): Implement DBSCAN Clustering
+* Week 7 (10/14): Integration and Testing Core Components
+* Week 8 (10/21): Integration and Testing Core Components
+* Week 9 (10/28): Develop API Components
+* Week 10 (11/04): Develop API Components
+* Week 11 (11/11): Build Pytyon Library
+* Week 12 (11/18): Create build scripts and CI/CD pipeline
```
Comment (yungyuc) [inline project/chris901003/README.rst]: You include CI and testing too late.  Please revise the plan to do it in very early stage to help you improve productivity.

**Review by yungyuc — CHANGES_REQUESTED** [review]
Log (plain-text) data analytics is not a perfect use case for the hybrid architecture that we cover for the numerical programming.  I recommend you to change a topic.  If you are not sure about what to do, please pick one from the example project list.

- [x] Please add an entry in `project/README.rst` pointing to your proposal.
- [x] Please change to use less than 80 characters in a line.  Long lines are hard to review and annotate.
- [x] Add testing and CI early for productivity.

File: project/chris901003/README.rst (hunk 1 above)
Comment (chris901003) [reply project/chris901003/README.rst]: When I am clustering each sentence, I will use numerical calculations.

File: project/chris901003/README.rst (hunk 2 above)
Comment (chris901003) [reply project/chris901003/README.rst]: I think my current goal should be to first handle small log files.

File: project/chris901003/README.rst (hunk 3 above)
Comment (chris901003) [reply project/chris901003/README.rst]: OK

**Comment by chris901003** [comment]
I'm ready for next review

File: project/chris901003/README.rst (hunk 1 above)
Comment (yungyuc) [reply project/chris901003/README.rst]: Please specify what computing/calculation you will use for analyzing or processing the log in your proposal.

File: project/chris901003/README.rst [hunk 4]
```diff
@@ -0,0 +1,103 @@
...
+Use GitHub Actions to verify that the build works correctly.
+
+Schedule
+========
+
+* Week 3 (09/16): Proposal Submission
+* Week 4 (09/23): Setup Development Environment
+* Week 5 (09/30): Implement FastText Sentence Vectorization
+* Week 6 (10/07): Implement DBSCAN Clustering
+* Week 7 (10/14): Create build scripts and CI/CD pipeline
```
Comment (yungyuc) [inline project/chris901003/README.rst]: It is still too late to add build scripts after implementing feature.  Please revise to make it earlier.

**Review by yungyuc — CHANGES_REQUESTED** [review]
When requesting a next review, please leave a global comment that summarizes your change.

Unfinished issues:

- [x] Please add an entry in `project/README.rst` pointing to your proposal.
- [x] Please change to use less than 80 characters in a line.  Long lines are hard to review and annotate.

New issues:

- [x] [L96](https://github.com/yungyuc/nsdhw_24au/pull/38/files#diff-8c7b8c5efa0f89ca3da48fb1bcc70d2b3706ca164b8fcd57e9cecd8a4000722fR96): Testing and CI are not early enough.

File: project/chris901003/README.rst (hunk 2 above)
Comment (yungyuc) [reply project/chris901003/README.rst]: Please specify the size of the log files you want to handle.

**Comment by yungyuc** [comment]
Please seriously consider to change a topic.  It is hard to make a good presentation using log analytics.  Your proposal does not explain how you can make it possible.

**Comment by chris901003** [comment]
I'm ready for next review
I have revised my proposal

File: project/chris901003/README.rst [hunk 5]
```diff
@@ -0,0 +1,134 @@
...
+add advanced packet arrival and queueing models 
+based on Poisson processes.
+
+****************************
+System Architecture
+****************************
+
+**Core Components**:
+
+1. Implements the Poisson process to simulate random packet arrivals.
```
Comment (yungyuc) [inline project/chris901003/README.rst:64]: What pseudo random number generator (PRNG) will you use?

Comment (yungyuc) [reply project/chris901003/README.rst:64]: How do you plan to test the results involving a random number?

**Review by yungyuc — CHANGES_REQUESTED** [review]
- [x] [L64](https://github.com/yungyuc/nsdhw_24au/pull/38/files#diff-8c7b8c5efa0f89ca3da48fb1bcc70d2b3706ca164b8fcd57e9cecd8a4000722fR64): Specify the PRNG.
- [x] [L64](https://github.com/yungyuc/nsdhw_24au/pull/38/files#diff-8c7b8c5efa0f89ca3da48fb1bcc70d2b3706ca164b8fcd57e9cecd8a4000722fR64): Make it clear how to test with random numbers.

File: project/chris901003/README.rst (hunk 5 above)
Comment (chris901003) [reply project/chris901003/README.rst:64]: I will ensure that the randomly generated numbers are fixed by using a fixed seed.
> I will use exponential distribution

How do you plan to test the results involving a random number?
> I will ensure that the randomly generated numbers are fixed by using a fixed seed.

**Comment by chris901003** [comment]
I'm ready for next review
1. use exponential distribution to generate random number
2. fixed seed for testing

File: project/README.rst [hunk 6]
```diff
@@ -1,46 +1,135 @@
-================
-NSD Term Project
-================
+PoissonNet
```
Comment (yungyuc) [inline project/README.rst]: You destroyed the project README.  Please fix it and remove bad commits from history (squashing and forcing push may be the easiest way to do it).

**Review by yungyuc — CHANGES_REQUESTED** [review]
Ouch.  Please fix your error.

**Comment by chris901003** [comment]
I'm ready for next review
I'm sorry about that, I've dealt with it.

**Comment by yungyuc** [comment]
Please squash to get rid of the offending change https://github.com/yungyuc/nsdhw_24au/pull/38/commits/c46762e9cc00e7f7d05742b472e3348afc59c508

When you update the PR, please also resolve the conflict.  (Not your fault, but a later PR should resolve the conflict from an earlier merge.)

**Comment by chris901003** [comment]
I'm ready for next review
I've fix the conflict and squash the commit

File: project/chris901003/README.rst (hunk 5 above)
Comment (yungyuc) [reply project/chris901003/README.rst:64]: > I will ensure that the randomly generated numbers are fixed by using a fixed seed.
> 
> I will use exponential distribution

Please add it in your proposal and leave an inline comment to point it to me.

> > How do you plan to test the results involving a random number?
> 
> I will ensure that the randomly generated numbers are fixed by using a fixed seed.

Ditto.

**Review by yungyuc — CHANGES_REQUESTED** [review]
* [x]  [L64](https://github.com/yungyuc/nsdhw_24au/pull/38/files#diff-8c7b8c5efa0f89ca3da48fb1bcc70d2b3706ca164b8fcd57e9cecd8a4000722fR64): Specify the PRNG.
* [x]  [L64](https://github.com/yungyuc/nsdhw_24au/pull/38/files#diff-8c7b8c5efa0f89ca3da48fb1bcc70d2b3706ca164b8fcd57e9cecd8a4000722fR64): Make it clear how to test with random numbers.

**Comment by chris901003** [comment]
I'm ready for next review
I've added it to proposal

**Review by yungyuc — APPROVED** [review]
LGTM.  Let me know when you want it to be merged.

**Comment by chris901003** [comment]
I'm ready to merge. Thanks!

**Pull request merged**